src/server/transaction.cc

```diff
@@ -609,7 +614,7 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   // If we're the head of tx queue (txq_ooo is false), we remove ourselves upon first invocation
   // and successive hops are run by continuation_trans_ in engine shard.
   // Otherwise we can remove ourselves only when we're concluding (so no more hops will follow).
-  if ((is_concluding || !txq_ooo) && sd.pq_pos != TxQueue::kEnd) {
+  if ((is_concluding || !is_ooo) && sd.pq_pos != TxQueue::kEnd) {
```

Comment:
I think that you changed behaviour here.
Now is_ooo is based on OUT_OF_ORDER
but what if this transaction is the head of tx queue? line 670 in engine_shard.cc 
here you will not remove it from head , line 671 set continuation_trans_ to head.

Reply:
maybe we can add a check that if we set continuation_trans_ than sd.pq_pos is TxQueue::kEnd

Reply:
good catch, I thought about this but decided it's ok, now I need to see why the dcheck fails.

Reply:
ok, reverted the logic, changed the names of the variables, hopefully it's more clear now